Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -848,7 +857,17 @@ void ContainerNode::childrenChanged(const ChildChange& change)
     if (change.source == ChildChange::Source::API && change.type != ChildChange::Type::TextChanged)
         document().updateRangesAfterChildrenChanged(*this);
 
-    invalidateNodeListAndCollectionCachesInAncestors();
+    if (change.affectsElements())
+        invalidateNodeListAndCollectionCachesInAncestors();
+    else if (change.type != ChildChange::Type::TextChanged) {
+        // Only the ChildNodeList cache is relevant when non-Elements are added / removed. Other types of
```

Comment:
I don't think this comment is useful. It's pretty self evident from the code.
If anything, I'd just say "Only ChildNodeList can contain text nodes".

Reply:
Ok, I'll drop.